static_frame/core/frame.py

```diff
@@ -5054,14 +5054,14 @@ def blocks() -> tp.Iterator[np.ndarray]:
         else:
             block_gen = blocks
 
-        if self._columns.depth > 1:
-            raise ErrorInitFrame('cannot unset index with a columns with depth greater than 1')
-
         if names:
             columns = chain(names, self._columns.values)
         else:
             columns = chain(self._index.names, self._columns.values)
 
+        if self._columns.depth > 1:
+            columns = IndexHierarchy.from_labels(columns)
+
```

Comment:
Here, we need to be pass on the same composed index constructors that are already used in the `Index`.

```suggestion
        if self._columns.depth > 1:
            index_constructors = tuple(self._index._levels.index_types())
            columns = IndexHierarchy.from_labels(columns, index_constructors=index_constructors)

```

Also, this PR suggests a different approach to ``blocks()``, above. This permits us to use the underlying arrays from the `IndexHierarchy` without doing type consolidation into a single array.

```python
        def blocks() -> tp.Iterator[np.ndarray]:
            if self._index.ndim == 1:
                yield self._index.values
            else:
                if self._index._recache:
                    self._index._update_array_cache()
                yield from self._index._blocks._blocks
            for b in self._blocks._blocks:
                yield b
```

Note that SF commonly uses private interfaces of component objects in functionality like this to avoid creating unnecessary intermediary data structures.